packages/core/src/shared/filetypes.ts

```diff
@@ -355,7 +355,7 @@ export const wellKnownCodeFiles = new Set(['Dockerfile', 'Dockerfile.build'])
 export function isCodeFile(filename: string): boolean {
     if (codefileExtensions.has(path.extname(filename).toLowerCase())) {
         return true
-    } else if (wellKnownCodeFiles.has(filename)) {
+    } else if (wellKnownCodeFiles.has(path.basename(filename))) {
         return true
     } else {
         return false
```

Comment:
would the lowercase step also be needed?

please also revert the change that you made in the previous pr, so that this code is less verbose:

```typescript
    const ext = path.extname(filename).toLowerCase()
    return codefileExtensions.has(ext) || wellKnownCodeFiles.has(path.basename(filename)
```

Reply:
I do think we want a case sensitive match in this case.  `Dockerfile` is the convention when used without an extension: https://docs.docker.com/build/concepts/dockerfile/#filename.

Reply:
That's the case for `Dockerfile`, but likely not other "filenames without an extension" . I guess we can wait until we encounter such files.